tests/Unit/Jobs/IndexBlocksTest.php

```diff
@@ -59,6 +67,51 @@
     });
 });
 
+it('should index new blocks using the timestamp from cache', function () {
+    Event::fake();
+
+    Cache::shouldReceive('get')
+        ->with('latest-indexed-timestamp:blocks')
+        ->andReturn(2) // so new ones are the one with timestamp 5 and 10
+        ->shouldReceive('put')
+        ->with('latest-indexed-timestamp:blocks', 10)
+        ->once();
+
+    Block::factory()->create([
+        'timestamp' => 10,
+    ]);
+
+    Block::factory()->create([
+        'timestamp' => 5,
+    ]);
+
+    Block::factory()->create([
+        'timestamp' => 1,
+    ]);
+
+    IndexBlocks::dispatch();
+
+    Event::assertDispatched(ModelsImported::class, function ($event) {
+        return $event->models->count() === 2;
```

Comment:
Is it worth stricter checks here (and the other tests that do the same) to make sure the correct blocks are pulled to be indexed?

```suggestion
        return $event->models->count() === 2 && $event->models[0]->timestamp === 5 && $event->models[1]->timestamp === 10;
```